backend/test/customerTests/controller.test.ts

```diff
@@ -0,0 +1,181 @@
+import * as sinon from 'sinon';
+import sinonStubPromise from 'sinon-stub-promise';
+import { expect } from 'chai';
+import request from 'supertest';
+import 'mocha';
+import * as bodyParser from 'body-parser';
+import jwt from 'jsonwebtoken';
+import { config } from '../../config';
+
+import App from '../../src/app';
+import CustomerService from '../../src/Customer/customer.service';
+import CustomerController from '../../src/Customer/customer.controller';
+
+sinonStubPromise(sinon);
+
+let sandbox: sinon.SinonSandbox;
+
+const testPort = 5001;
+
+const payload = {
+    userId: 1,
+    name: 'Fake Name',
+    role: 'Fake Role'
+};
+
+const mockCustomer = {
+    name: 'john',
+    email: 'doe'
+};
+
+const badMockCustomer = {
+    name: 'john',
+    email: undefined
+};
+
+const archiveCustomer = {
+    id: 1,
+    archived: true
+};
+
+const token = jwt.sign(payload, config.jwt_public_key, { expiresIn: '1d' });
+
+describe('Customer Controller Test', () => {
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it('Test get all customers', async () => {
+        const mockCustomerService = sandbox.createStubInstance(CustomerService);
+        mockCustomerService.getAllCustomers.resolves('foo');
+        const app = new App({
+            port: testPort,
+            controllers: [new CustomerController(mockCustomerService)],
+            middleWares: []
+        });
+
+        app.listen();
+        const res = await request(app.app)
+            .get('/customer/')
+            .set('Accept', 'application/json')
+            .set('Authorization', 'bearer ' + token);
+        expect(res.body).to.equal('foo');
+        expect(res.status).to.equal(200);
+        app.shutdown();
+    });
+
+    it('Test create new customer route', async () => {
```

Comment:
Could add a test where the customer already exists and assert that there is an error.

So for that add an error handler when we try to create a customer that already exist:
https://github.com/Talos10/SOEN390-team-9/blob/main/backend/src/Customer/customer.service.ts (line 15)

You can refer to this issue of the problem: #106 
It is the same issue but for users, if you can fix #106 too it would be awesome.

Reply:
We don't make customers unique the same way we do for the users and so we won't have a test like that for customers.

And yes, I will fix #106 in this PR.